ZelBack/src/services/appsService.js

```diff
@@ -10158,11 +10159,80 @@ async function trySpawningGlobalApplication() {
     }
 
     // an application was selected and checked that it can run on this node. try to install and run it locally
+    // lets broadcast to the network the app is going to be installed on this node, so we don't get lot's of intances installed when it's not needed
+    const broadcastedAt = Date.now();
+    const newAppRunningMessage = {
+      type: 'fluxapprunning',
+      version: 1,
+      name: appSpecifications.name,
+      hash: appSpecifications.hash, // hash of application specifics that are running
+      ip: myIP,
+      broadcastedAt,
+      runningSince: broadcastedAt,
+      osUptime: os.uptime(),
+      staticIp: geolocationService.isStaticIP(),
+    };
+
+    // store it in local database first
+    // eslint-disable-next-line no-await-in-loop, no-use-before-define
+    await storeAppRunningMessage(newAppRunningMessage);
+    // broadcast messages about running apps to all peers
+    await fluxCommunicationMessagesSender.broadcastMessageToOutgoing(newAppRunningMessage);
+    await serviceHelper.delay(500);
+    await fluxCommunicationMessagesSender.broadcastMessageToIncoming(newAppRunningMessage);
+    // broadcast messages about running apps to all peers
+
+    await serviceHelper.delay(30 * 1000); // give it time so messages are propagated on the network
```

Comment:
This is just a question... have we done any modeling around how long it takes for a message to propagate through the network?

Reply:
I have done a manualy check long time ago, it would take couple of seconds. We haven't done any modeling/professional check :D